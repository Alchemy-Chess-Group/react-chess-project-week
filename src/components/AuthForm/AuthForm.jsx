import React from 'react';
import { useState } from 'react';
import { useHistory } from 'react-router-dom';
import style from './AuthForm.css';

export default function AuthForm({ handleAuth }) {
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');

  const handleSubmit = async (e) => {
    e.preventDefault();
    await handleAuth(email, password);
  };

  return (
    <form className={style.authForm}>
      <label>
        <input
          onChange={(e) => setEmail(e.target.value)}
          value={email}
          type="email"
          placeholder="email"
          required="required"
        />
      </label>

      <label>
        <input
          onChange={(e) => setPassword(e.target.value)}
          value={password}
          type="password"
          placeholder="password"
          required="required"
        />
      </label>
      <button onClick={handleSubmit}>submit</button>
    </form>
  );
}
