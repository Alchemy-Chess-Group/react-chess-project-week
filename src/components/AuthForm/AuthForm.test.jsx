import { render, screen } from '@testing-library/react';
import AuthForm from './AuthForm';

test('auth page should render a form', () => {
  const { container } = render(<AuthForm />);

  screen.getByRole('textbox', { label: /email/i });

  screen.getByRole('textbox', { label: /password/i });

  screen.getByRole('button', { name: /submit/i });

  expect(container).toMatchSnapshot();
});
