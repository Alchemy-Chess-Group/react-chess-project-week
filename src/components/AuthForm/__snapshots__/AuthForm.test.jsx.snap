// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`auth page should render a form 1`] = `
<div>
  <form
    class="authForm"
  >
    <label>
      <input
        placeholder="email"
        required=""
        type="email"
        value=""
      />
    </label>
    <label>
      <input
        placeholder="password"
        required=""
        type="password"
        value=""
      />
    </label>
    <button>
      submit
    </button>
  </form>
</div>
`;
